src/game/common/system/localfile.cpp

```diff
@@ -103,227 +103,198 @@ bool LocalFile::Open(const char *filename, int mode)
 }
 
-int LocalFile::Seek(int offset, File::SeekMode mode)
+int LocalFile::Seek(int offset, SeekMode mode)
 {
     switch (mode) {
-        case SeekMode::START:
-        case SeekMode::CURRENT:
-        case SeekMode::END:
+        case START:
+            return lseek(m_fileHandle, offset, START);
+        case CURRENT:
+            return lseek(m_fileHandle, offset, CURRENT);
+        case END:
+            captainslog_dbgassert(
+                offset <= 0, "LocalFile::Seek - pos should be <= 0 for a seek starting at the end of the file");
+            return lseek(m_fileHandle, offset, END);
             break;
         default:
             return -1;
     }
-
-    return lseek(m_fileHandle, offset, mode);
 }
 
 void LocalFile::Next_Line(char *dst, int bytes)
 {
-    captainslog_trace("Seeking getting next line from LocalFile %s.", m_filename.Str());
+    char tmp = 0;
+    int pos = 0;
+    int count;
 
-    int i;
-
-    for (i = 0; i < bytes - 1; ++i) {
-        char tmp;
-        int ret = read(m_fileHandle, &tmp, sizeof(tmp));
-
-        if (ret <= 0 || tmp == '\n') {
-            break;
+    do {
+        if (dst == nullptr || pos >= bytes - 1) {
+            count = read(m_fileHandle, &tmp, 1);
+        } else {
+            count = read(m_fileHandle, &dst[pos], 1);
+            tmp = dst[pos];
         }
 
-        if (dst != nullptr) {
-            dst[i] = tmp;
-        }
-    }
+        pos++;
+    } while (count != 0 && tmp != '\n');
 
     if (dst != nullptr) {
-        dst[i] = '\0';
+        if (pos < bytes) {
+            dst[pos] = 0;
+        } else {
+            dst[bytes] = 0;
+        }
     }
 }
 
 bool LocalFile::Scan_Int(int &integer)
 {
-    captainslog_trace("Scanning Int from LocalFile %s.", m_filename.Str());
-    char tmp;
-    Utf8String number;
-
     integer = 0;
+    Utf8String number;
+    int count;
+    char tmp;
 
-    // Loop to find the first numeric character.
     do {
-        if (read(m_fileHandle, &tmp, sizeof(tmp)) == 0) {
-            return false;
-        }
-    } while (!isdigit(tmp) && tmp != '-');
+        count = read(m_fileHandle, &tmp, 1);
+    } while (count != 0 && (tmp < '0' || tmp > '9') && tmp != '-');
 
-    // Build up our string of numeric characters
-    while (true) {
-        number.Concat(tmp);
+    if (count == 0) {
+        return false;
+    } else {
+        do {
+            number.Concat(tmp);
+            count = read(m_fileHandle, &tmp, 1);
+        } while (count != 0 && tmp >= '0' && tmp <= '9');
 
-        if (read(m_fileHandle, &tmp, sizeof(tmp)) == 0) {
-            break;
+        if (count != 0) {
+            lseek(m_fileHandle, -1, CURRENT);
         }
 
-        if (!isdigit(tmp)) {
-            // If we read a byte, seek back that byte for the next read
-            // as we are done with the current number.
-            lseek(m_fileHandle, -1, SeekMode::CURRENT);
-
-            break;
-        }
+        integer = atoi(number);
```

Comment:
number.Str()

Reply:
There is no more need to call .Str here than there is in any of the hundreds of other places across our codebase (including many places elsewhere in the file code) where we rely on Utf8String::operator const char *

If we no longer want to use that operator then we should get rid of it completely and add .Str across the entire codebase but that's for a different PR.